### ISSUE TITLE
sentry_sdk does not honor environment tag set in events or scope

### ISSUE THREAD

**Issue by TeddyHartanto** [getsentry/sentry-python#1466]
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.5.12

### Steps to Reproduce

1. Initialise `sentry_sdk` with environment=X
2. Send an event with a tag environment=Y, where Y != X.

Expectation: event should be sent to environment Y. In reality, it is sent to environment X.

### Expected Result

Events tagged with `environment` should overwrite the `environment` determined during the initialisation of the SDK

### Actual Result

The snippet below tag messages with the correct environment (`staging`):
```python
>>> sentry_sdk.init(dsn=SENTRY_DSN, environment="staging")
<sentry_sdk.hub._InitGuard object at 0x1022a76d0>
>>> sentry_sdk.capture_message("Test by Teddy")
'7c8556486f3a4f1692d6dc553a184992'
```

However, when sdk is initialised without setting the environment argument (defaults to `production`), sending messages tag with `environment` does not seem to work. I expect some of the environment-tagged messages below to go to `staging`, but they all go to `production`.
```python
>>> sentry_sdk.init(dsn=SENTRY_DSN)
<sentry_sdk.hub._InitGuard object at 0x10135e700>
>>> sentry_sdk.capture_message("Test by Teddy")  # goes to production
'b8b75d6570754bdaaa5a5cdbd2ada6f5'
>>> sentry_sdk.capture_message("Test by Teddy", tags=dict(environment="staging"))    # goes to production, should have been staging
'412230f90a124331b5e406ff15979187'
>>> sentry_sdk.set_tag("environment", "staging")
>>> sentry_sdk.capture_message("Test by Teddy")   # goes to production, should have been staging
'b42eca378d2641ec8f7d11151b81ee4c'
```

Likewise, in the below scenario, `sentry_sdk` only honors the `environment` argument passed during its initialisation.
```python
>>> sentry_sdk.init(dsn=SENTRY_DSN, environment="staging")
<sentry_sdk.hub._InitGuard object at 0x1025727f0>
>>> sentry_sdk.capture_message("Test by Teddy", tags=dict(environment="production"))  # goes to staging, should have been production
'30fd2a2fa4994dfcb882d7c44ffb1ae7'
```

That seems to be a discrepancy with what was documented in the [doc](https://docs.sentry.io/platforms/python/configuration/environments/):
> Sentry automatically creates an environment when it receives an event with the environment tag set.

"**when it receives an event with the environment tag set**" implies that we **can** tag events with an environment

What is the truth here? Is that behaviour expected? Is the doc outdated or is this a bug?

**Comment by antonpirker** [comment]
Hey @TeddyHartanto !

Thanks for writing in!
`environments` and `tags` are different things in Sentry so it is not possible to change the environment by setting a tag.

The docs are misleading. It should be read as:

> Sentry automatically creates an environment when it receives an event with the environment parameter set.

I created a PR in the docs to fix this:
https://github.com/getsentry/sentry-docs/pull/5185

**Comment by TeddyHartanto** [comment]
Thank you for the clarification @antonpirker.

I just saw another instance of the same sentence here: https://docs.sentry.io/product/sentry-basics/environments/#creating-environments

FYI, I just want to add that I might have gotten the idea of setting an `environment` tag because of what I saw in the UI too:
![image](https://user-images.githubusercontent.com/11240927/174957892-63309c73-988d-420e-b011-29a301ff6a25.png)

I think presenting `environment` as a component in the `tags` could create some confusion. It would be great if we can clarify in the docs that perhaps `environment` is a special tag that is only set in the `init()` function.
